src/fsharp/service/service.fs

```diff
@@ -1855,22 +1855,23 @@ type FSharpCheckProjectResults(projectFileName:string, tcConfigOption, keepAssem
     member info.GetUsesOfSymbol(symbol:FSharpSymbol) = 
         let (tcGlobals, _tcImports, _thisCcu, _ccuSig, tcSymbolUses, _topAttribs, _tcAssemblyData, _ilAssemRef, _ad, _tcAssemblyExpr, _dependencyFiles) = getDetails()
 
-        [| for r in tcSymbolUses do yield! r.GetUsesOfSymbol(symbol.Item) |] 
-        |> Seq.distinctBy (fun (itemOcc,_denv,m) -> itemOcc, m) 
-        |> Seq.filter (fun (itemOcc,_,_) -> itemOcc <> ItemOccurence.RelatedText) 
-        |> Seq.map (fun (itemOcc,denv,m) -> FSharpSymbolUse(tcGlobals, denv, symbol, itemOcc, m)) 
+        tcSymbolUses
+        |> Seq.collect (fun r -> r.GetUsesOfSymbol symbol.Item)
+        |> Seq.distinctBy (fun symbolUse -> symbolUse.ItemOccurence, symbolUse.Range) 
+        |> Seq.filter (fun symbolUse -> symbolUse.ItemOccurence <> ItemOccurence.RelatedText) 
+        |> Seq.map (fun symbolUse -> FSharpSymbolUse(tcGlobals, symbolUse.DisplayEnv, symbol, symbolUse.ItemOccurence, symbolUse.Range)) 
         |> Seq.toArray
         |> async.Return
 
     // Not, this does not have to be a SyncOp, it can be called from any thread
     member info.GetAllUsesOfAllSymbols() = 
```

Comment:
This was always a really slow function, I often wondered if an additional filter could be used for the consumers to limit the returned symbols.  Its used in semantic highlighting in vs4mac and in ionide too for codelens I think.

Reply:
filter to what?

Reply:
In this issue I noticed that we're getting multiple symbols for the same range:  https://github.com/Microsoft/visualfsharp/issues/4231

Perhaps that's something worth filtering on, but I think it's more complicated than that.

Reply:
Well I looked at the call sites and they do distinctby. Maybe we need to check these